site/student-lectures/Search-Space-Analysis/Search Space Analysis.rst

```diff
@@ -0,0 +1,208 @@
+**********************************************
+Search Space Analysis in Computational Systems
+**********************************************
+
+Introduction to Search Spaces
+----------------------------
+
+Search Space Terminology 
+------------------------
+* search space size #S
+* number of (optimal) solutions #S' ,solution density #S'/#S
+* search space diameter (:math:`G_{N}`) = maximal distance between any two candidate solutions
+* N(:math:`pi`)  is the  neighbourhood relation of a given candaicae slution (:math:`pi`)
```

Comment:
did you want this to be p_i? It is fine if not; I'm just asking.

Reply:
1.nope. the paper I saw used pi (greek letter) so I was using the same notation as them